src/madness/world/worldrmi.cc

```diff
@@ -332,10 +335,19 @@ namespace madness {
     void RMI::RmiTask::huge_msg_handler(void *buf, size_t /*nbytein*/) {
         const size_t* info = (size_t *)(buf);
         int nword = HEADER_LEN/sizeof(size_t);
-        int src = info[nword];
-        size_t nbyte = info[nword+1];
-
-        RMI::task_ptr->hugeq.push_back(std::make_pair(src,nbyte));
+        const int src = info[nword];
+        const size_t nbyte = info[nword+1];
+        const int tag = info[nword+2];
+
+        // extra dose of paranoia: assert that we never process so many huge messages
+        // that the tag wraparound somewhere becomes possible ...
+        // the worst case is where only one node sends huge messages to every node in the communicator
+        // AND it has enough threads to use up all tags
+        // NB list::size() is O(1) in c++11, but O(N) in older libstdc++
+        MADNESS_ASSERT(ThreadPool::size() < RMI::RmiTask::unique_tag_period() ||
+                       RMI::task_ptr->hugeq.size() <
+                           RMI::RmiTask::unique_tag_period() / comm.Get_size());
```

Comment:
@evaleev added another range for for huge message queue that is only used for huge messages. So there is no overlap with other uses.

As the check stands now, it does not do anything useful due to the or in the conditional expression. It will always be true when `ThreadPool::size() < RMI::RmiTask::unique_tag_period()`, which will be constant for the duration of the the run. It is highly unlikely that we will have that many threads for the foreseeable future.

If we remove the first half of the check (recommended), then the second half of the check will *ALWAYS* fail when the number processes is greater than the period.

The check should either be removed to avoid confusion, or fixed.